Artifacts/MsSql/Data/Security/1060-Publishing-ResourceClaims.sql

```diff
@@ -0,0 +1,87 @@
+BEGIN
```

Comment:
Should this file be under a Publishing subtype folder?  Right now its going to set up security even if the feature isn't enabled.

Reply:
Things have been moving around and changing so much, I didn't know that it was possible to add security metadata under a sub-type folder (even though this really isn't a sub-type, it really is more akin to an extension).

I didn't see any other places where we do this that I could use as a reference. Are there examples or documentation on this that I should be referencing?

Reply:
So I was going to mention Record Ownership... but it looks like we didn't do that one using subtypes either.  If you look in Deployment.psm1, all database reset tasks (including Ods, Admin, Security) pass along the subTypeNames in order to trigger sub type scripts if needed.  So its not currently being used, but is available to use.  In this case, we're actually probably fine to leave this in the main folder, so the supporting tables are always available for the feature, making it just a config setting change instead of also a database update to turn the feature on.

Reply:
Well, as it stands, the table creation scripts are implemented as sub-type on the Ed-Fi Standard (which was odd, but the only way to get them to run as "not an extension"). This script will cause the security metadata to be created even if the tables have not been created.

I'm fine with doing whatever is desired here. (My only thought here is that "features" should be treated as a higher-level abstraction of both extensions and features as they are implemented now (which only allows feature database integration as a sub-type of the Ed-Fi Standard and/or Ed-Fi extensions).

Reply:
Strange, for features it should still be supported to run them under Artifacts folders at the root of the repo (Ed-Fi-ODS for core supported features, Ed-Fi-ODS-Implementation for custom features from the field).  You can use the same approach there for scripts that should run more generally and aren't necessarily linked to a specific data model.  Have you tried in those folders?  I think that would match with what you mention, having a higher level location for scripts outside the data models.

Reply:
Actually, I removed this file completely because since it was converted from a feature+extension to just a feature, there is no participation in the authorization architecture as a securable resource anymore (for the _publishing/snapshots_ resource) -- all authenticated API clients will be able to see the available snapshot identifiers.